### PR TITLE
Tokenomics page

### PR DESCRIPTION
**Pull request by TopETH**

### CODE REVIEW

**Review by mogmachine — APPROVED** [review]

File: src/components/tokenomics/HalveningChart.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+			borderColor: "#fff",
+			label: {
+				style: {
+					color: "#000",
+				},
+				text: "Kusangi Network",
+			},
+		},
+		{
+			x: new Date("12 Mar 2021").getTime(),
```
Comment (cuteolaf) [inline src/components/tokenomics/HalveningChart.tsx:46]: Is it March or May?

File: src/components/tokenomics/HalveningChart.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+	const events = [
+		{
+			x: new Date("09 Jan 2021").getTime(),
+			borderColor: "#fff",
+			label: {
+				style: {
+					color: "#000",
+				},
+				text: "Kusangi Network",
```
Comment (cuteolaf) [inline src/components/tokenomics/HalveningChart.tsx:42]: It is **Kusanagi**, not **Kusangi**.

File: src/components/tokenomics/HalveningChart.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+		},
+		{
+			x: new Date("21 Nov 2021").getTime(),
+			borderColor: "#fff",
+			label: {
+				style: {
+					color: "#000",
+				},
+				text: "Nakamoto Network",
+			},
```
Comment (cuteolaf) [inline src/components/tokenomics/HalveningChart.tsx:63]: repeated several times.
refactor please.

File: src/components/tokenomics/HalveningChart.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+			x: "05/12/2021",
+			y: 546113,
+		},
+		{
+			x: "11/21/2021",
+			y: 546113,
+		},
+		{
+			x: "05/08/2024",
+			y: 6727871,
```
Comment (cuteolaf) [inline src/components/tokenomics/HalveningChart.tsx:30]: Where did you copy this value from?

File: src/screens/tokenomics.tsx [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+	const totalSupply = rawAmountToDecimal("21000000000000000");
+	const circulatingSupply = rawAmountToDecimal(chain.issued.toString());
+	const unissued = totalSupply.minus(circulatingSupply);
+	const stakedCirculatingSupply = rawAmountToDecimal(chain.staked.toString());
+	const freeCirculatingSupply = circulatingSupply.minus(
+		stakedCirculatingSupply
+	);
+	const circulationPercent = circulatingSupply.mul(100).div(totalSupply);
+	const blocksPerHalvening = 10500000;
+	const secondsPerHalvening = 126000000;
```
Comment (cuteolaf) [inline src/screens/tokenomics.tsx:80]:
```suggestion
	const secondsPerHalvening = blocksPerHalvening  * 12;
```

File: src/screens/tokenomics.tsx [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+			issue: "10,253.90625",
+			reward: "0.00048828125",
+		},
+		{
+			label: "H12",
+			total: "20,994,873.046875",
+			issue: "5,126.953125",
+			reward: "0.000244140625",
+		},
+	];
```
Comment (cuteolaf) [inline src/screens/tokenomics.tsx:165]: Does it need to be hard coded?

File: src/components/tokenomics/TokenomicsChart.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	const strStaked = `Circulating Delegated/Staked (${formatNumber(
+		stakedPercent,
+		{ decimalPlaces: 2 }
+	)}% of ${nFormatter(circulatingSupply.toNumber(), 2)})`;
+	const strFree = `Circulating Free (${formatNumber(freePercent, {
+		decimalPlaces: 2,
+	})}% of ${nFormatter(circulatingSupply.toNumber(), 2)})`;
+	const strUnissued = `Unissued (${formatNumber(unissuedPercent, {
+		decimalPlaces: 2,
+	})}% of ${nFormatter(totalSupply.toNumber(), 2)})`;
```
Comment (cuteolaf) [inline src/components/tokenomics/TokenomicsChart.tsx:39]: looks a bit messy, please refactor.

**Review by cuteolaf — CHANGES_REQUESTED** [review]

File: src/components/tokenomics/HalveningChart.tsx (hunk 1 above)
Comment (TopETH) [reply src/components/tokenomics/HalveningChart.tsx:46]: It's March (in Taostats)

File: src/components/tokenomics/HalveningChart.tsx (hunk 2 above)
Comment (TopETH) [reply src/components/tokenomics/HalveningChart.tsx:42]: Fixed

File: src/components/tokenomics/HalveningChart.tsx (hunk 3 above)
Comment (TopETH) [reply src/components/tokenomics/HalveningChart.tsx]: Fixed

File: src/components/tokenomics/HalveningChart.tsx (hunk 1 above)
Comment (cuteolaf) [reply src/components/tokenomics/HalveningChart.tsx]: Check again.

File: src/components/tokenomics/HalveningChart.tsx (hunk 4 above)
Comment (TopETH) [reply src/components/tokenomics/HalveningChart.tsx:30]: Fixed

File: src/screens/tokenomics.tsx (hunk 5 above)
Comment (TopETH) [reply src/screens/tokenomics.tsx]: Fixed

File: src/components/tokenomics/TokenomicsChart.tsx (hunk 7 above)
Comment (TopETH) [reply src/components/tokenomics/TokenomicsChart.tsx]: Fixed

File: src/components/tokenomics/HalveningChart.tsx (hunk 1 above)
Comment (TopETH) [reply src/components/tokenomics/HalveningChart.tsx]: Fixed

**Review by cuteolaf — APPROVED** [review]

**Pull request merged**